### ISSUE TITLE
Question: Is localhost handled differently?

### ISSUE THREAD

**Issue by pwright** [rlhk/logseq-url-plus#6]
When I run an app locally with a restful api locally and hit it with `URL+ API -> JSON Code`, it doesn't return anything
Is there something special about localhost? 

Console error is:

```
index.js:727 Uncaught (in promise) Error: link-preview-js did not receive a valid a url or text
    at Object.<anonymous> (index.js:727:398)
    at B (index.js:725:503)
    at Object.next (index.js:724:306)
    at index.js:724:215
    at new Promise (<anonymous>)
    at t (index.js:724:8)
    at e.getLinkPreview (index.js:727:87)
    at index.js:1458:325
    at index.js:1196:131
```

**Comment by rlhk** [comment]
Occasionally I have to try again if there no response.

**Comment by rlhk** [comment]
Oh, you mean localhost api. I don't know actually. Let me setup a local api and check later

**Comment by rlhk** [comment]
Not sure if it's related to this issue: https://stackoverflow.com/questions/54164874/javascript-use-fetch-on-localhost-with-url-parameters

But I think the plugin code could be optimised a bit. It tries to retrieve metadata via link-preview-js even though it's not needed for API response. Shall fix next.

**Comment by rlhk** [comment]
The plugin relies on `link-preview-js` to retrieve URL metadata. Apparently it doesn't handle `localhost` request. The issue is fixed by not calling it on API request. Should be fixed in 0.0.7 release. 

Note that `127.0.0.1` will cause CORS error if server is not setup, so better avoid.

**Comment by pwright** [comment]
wow, thanks for the quick response @rlhk - works perfectly
